src/mongocxx/client.hpp

```diff
@@ -226,7 +226,11 @@ class MONGOCXX_API client {
     MONGOCXX_PRIVATE explicit client(void* implementation);
 
     class MONGOCXX_PRIVATE impl;
-    std::unique_ptr<impl> _impl;
+
+    MONGOCXX_PRIVATE impl& _get_impl();
+    MONGOCXX_PRIVATE const impl& _get_impl() const;
+
+    MONGOCXX_PRIVATE std::unique_ptr<impl> _impl;
```

Comment:
nit: you don't need the MONGOCXX_PRIVATE on members. Only on types and functions.